File: Java8-Projects/JavaSkill/src/com/app/main/MainApp.java
package com.app.main;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;
import java.util.Map;

class A {
	A(){
		System.out.println("A");
	}
}
public class MainApp extends A{
	MainApp(){
		System.out.println("B");
	}

	static Boolean b1[] = new Boolean[5];

	public static void main(String[] args) {
//		String x = new String("test");
//		String y = "test";
//		System.out.println(x == y);
//		System.out.println(x.equals(y));


//		 System.out.println(b1[0]);
		float a = 10.20f;
		float b = 10.20f;

		new MainApp();


	}
	public static void show(Object o){
		System.out.println("Hello");
	}
}

File: Kotlin-Projects/kotlin-mysql-app/src/main/kotlin/com/app/kolin/MainApp.kt
package com.app.kolin

import com.app.entities.Book
import com.app.entities.DataBook
import com.app.practical.HashMapFiltering
import com.app.practical.Sorting
import java.util.stream.Collectors
import java.util.Comparator
import java.util.stream.Collector
import java.util.stream.Stream
import kotlin.streams.toList


fun main() {

//    var dbook = DataBook(1, "Java", "Pob", 100.0, 20)
//    var copyObj = dbook.copy()
//    println(dbook == copyObj) //is true
//    println(dbook === copyObj) //is false
//    println(dbook.equals(copyObj)) //is true

//    newDbook.authors = "Jackson"

//    println(newDbook.toString())
//    println(dbook.toString())

//    Sorting()
//    HashMapFiltering()


    // Converting from hashMap to ArrayList

//    var book = Book()
//    var bookList = book.findAllByBookList()
//    println(bookList)

//    val names = bookList
//        .asSequence()
//        .filter { it.price > 350 && it.price < 400}
//        .map { it }
//        .distinct()
//        .sortedBy { book -> book.id }
//        .toList()
//
//    var values : String = ""
//    for (i in names.withIndex()) println("$values")

    val arrayOfEmps = arrayOf<DataBook>(
        DataBook(30, "Java8", "Joe", 200.0, 20)
    )
    var dataBookList = Stream.of(arrayOfEmps).collect(Collectors.toList())

    println(arrayOfEmps)

//    val strings = Stream.of("how", "to", "do", "in", "java").collect(Collectors.toList())
//    println(strings)


//    val keyList = ArrayList(listMap[0].keys)
//    val valueList = ArrayList(listMap[0].values)
//    println("$keyList, $valueList")


}